File: Sorting/QuickSort.py
def QuickSort(arr, low, high):
    if low < high:
        pivot = partition(arr, low, high)

        QuickSort(arr, low, pivot - 1)
        QuickSort(arr, pivot + 1, high)

def partition(arr, low, high):
    pivot = arr[high]
    l = low - 1
    for i in range(low,high):
        if arr[i] <= pivot:
            l += 1
            t = arr[i]
            arr[i] = arr[l]
            arr[l] = t
    l += 1
    t = arr[high]
    arr[high] = arr[l]
    arr[l] = t
    return l

arr = [2,1,5,9,3,3,6]
print(arr)
QuickSort(arr, 0, len(arr)-1)
print(arr)

File: Sorting/QuickSort.cpp
#include<iostream>
#include<vector>
using namespace std;
int partition(vector<int> &arr, int low, int high){
    int pivot = arr[high];
    int j = low - 1;
    for(int i = low; i<high; i++){
        if(arr[i] <= pivot){
            j+=1;
            int t = arr[i];
            arr[i] = arr[j];
            arr[j] = t;
        }
    }
    int t = arr[high];
    arr[high] = arr[j+1];
    arr[j+1] = t;
    return j+1;
}
void QuickSort(vector<int> &arr, int low, int high){
    if(low < high){
        int pivot = partition(arr,low,high);
        QuickSort(arr, low , pivot-1);
        QuickSort(arr, pivot+1, high);
    }
}

int main(){
    vector<int> arr = {2,5,1,9,4,3};
    QuickSort(arr, 0,arr.size()-1);
    for(int ele : arr){
        cout<<ele<<", ";
    }
}